app.ts

```diff
@@ -323,6 +338,89 @@ export default class MELCloudApp extends App {
+    this.outdoorTemperatureListener = outdoorTemperatureDevice.makeCapabilityInstance(
+      outdoorTemperatureCapability,
+      async (): Promise<void> => {
+        try {
+          const [outdoorTemperatureDevice, outdoorTemperatureCapability] = await this.extractOutdoorTemperatureData(capabilityPath)
+          const outdoorTemperature: number = outdoorTemperatureDevice.capabilitiesObj[outdoorTemperatureCapability].value
+          for (const device of this.getDevices({ driverId: 'melcloud' })) {
+            if (device.getCapabilityValue('operation_mode') !== 'cool') {
+              continue
+            }
+            await device.onCapability('target_temperature', Math.max(threshold, outdoorTemperature - 8))
+          }
+        } catch (error: unknown) {
+          this.error(error instanceof Error ? error.message : error)
+          this.destroyOutdoorTemperatureListener(true)
+        }
+      }
+    )
+  }
+
+  async isNewCapabilityInstanceNeeded ({ capabilityPath, enabled, threshold }: OutdoorTemperatureListenerData): Promise<boolean> {
+    let haveSettingsChanged: boolean = false
+    if (enabled && (capabilityPath === '' || threshold === 0)) {
+      throw new Error('Outdoor temperature and/or threshold are missing.')
+    }
+    if (enabled !== this.homey.settings.get('self_adjust_enabled')) {
+      this.homey.settings.set('self_adjust_enabled', enabled)
+      haveSettingsChanged = true
+    }
+    if (capabilityPath !== this.homey.settings.get('outdoor_temperature_capability_path')) {
+      this.homey.settings.set('outdoor_temperature_capability_path', capabilityPath)
+      haveSettingsChanged = true
+    }
+    if (threshold !== this.homey.settings.get('self_adjust_threshold')) {
+      this.homey.settings.set('self_adjust_threshold', threshold)
+      haveSettingsChanged = true
+    }
+    if (!enabled || haveSettingsChanged) {
+      this.destroyOutdoorTemperatureListener()
+    }
+    return enabled && haveSettingsChanged
+  }
+
+  async extractOutdoorTemperatureData (capabilityPath: string): Promise<[any, string]> {
+    try {
+      const [id, outdoorTemperatureCapability]: string[] = capabilityPath.split(':')
+      // @ts-expect-error bug
+      const outdoorTemperatureDevice = await this.api.devices.getDevice({ id })
+      if (!(outdoorTemperatureCapability in outdoorTemperatureDevice.capabilitiesObj)) {
+        throw new Error('not_found')
+      }
+      return [outdoorTemperatureDevice, outdoorTemperatureCapability]
+    } catch (error: unknown) {
+      this.destroyOutdoorTemperatureListener(true)
+      throw new Error(String(error))
+    }
+  }
+
+  destroyOutdoorTemperatureListener (resetData: boolean = false): void {
+    if (resetData) {
+      this.homey.settings.set('outdoor_temperature_capability_path', '')
+      this.homey.settings.set('self_adjust_enabled', false)
+    }
+    if (this.outdoorTemperatureListener !== null) {
+      this.outdoorTemperatureListener.destroy()
```

Comment:
Hi @robertklep, if I well understand HomeyKitty, it would be necessary to add
```
deviceRelatedToOutdoorTemperatureListener.io = null
```
as a workaround in order to make any `outdoorTemperatureListener` work properly, is that right?

Reply:
Setting the `io` property to `null` is only required if you want to call `makeCapabilityInstance` more than once for the same device/capability combination (for instance with HomeKitty, when you remove a device from HomeKit and then add it back again). See also [this issue](https://github.com/athombv/homey-web-api-issues/issues/22) that explains it in a bit more detail.

Reply:
Ok, thanks @robertklep (indeed I've just noticed you put the reference to this issue in your code...).

So I don't think it applies to me since the `api.devices.getDevice({ id })` is retrieved from API in case the listener should be recreated after having been destroyed.

By the way I created this issue: https://github.com/athombv/homey-web-api-issues/issues/23, but I feel like the web API issues aren't really being addressed.

Reply:
You have to be careful that the device you retrieve isn't a cached instance, in which case you may still run into the issue.

As for Athom dealing with Web API issues: some are dealt with quickly, but there's only one developer working on them (from what I can see) and it's a lot of legacy code.

Reply:
You were right @robertklep, device is cached from `id`.
So fixed [here](https://github.com/OlivierZal/com.melcloud/pull/80/commits/f6bc0438279606f489316f88deb111af3d39bea1#diff-84dd6a83459413d5bb638cd885f20ef0dbebf2ebf10d0f35d1c9ca424b0ca06aR418).

Thank you very much.